From 09d16b547abdd92e89d143619e186149b159cee5 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:39 -0400
Subject: [powerpc] Check "status" property before adding legacy ISA serial ports

Message-id: <1410545655-205645-51-git-send-email-dzickus@redhat.com>
Patchwork-id: 94154
O-Subject: [RHEL7 PATCH 050/626] powerpc: Check "status" property before adding legacy ISA serial ports
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit e0f5fa99a378d6910d234f3c8998aebfe2f54745
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Mon Jul 15 13:03:13 2013 +1000

    powerpc: Check "status" property before adding legacy ISA serial ports

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/legacy_serial.c b/arch/powerpc/kernel/legacy_serial.c
index afbc535..22e88dd 100644
--- a/arch/powerpc/kernel/legacy_serial.c
+++ b/arch/powerpc/kernel/legacy_serial.c
@@ -386,9 +386,11 @@ void __init find_legacy_serial_ports(void)
   struct device_node *isa = of_get_parent(np);
   if (isa && (!strcmp(isa->name, "isa") ||
        !strcmp(isa->name, "lpc"))) {
-   index = add_legacy_isa_port(np, isa);
-   if (index >= 0 && np == stdout)
-    legacy_serial_console = index;
+   if (of_device_is_available(np)) {
+    index = add_legacy_isa_port(np, isa);
+    if (index >= 0 && np == stdout)
+     legacy_serial_console = index;
+   }
   }
   of_node_put(isa);
  }
-- 
1.7.1